projects/step-frontend/src/lib/modules/timeseries/modules/_common/components/time-range-picker/time-range-picker.component.ts

```diff
@@ -48,6 +47,13 @@ export class TimeRangePickerComponent implements OnInit, OnChanges {
   // when auto-refresh is enabled or the changes come from exterior, the inputs may be updated in the middle of editing
   dateTimeInputsLocked = false;
 
+  activeSelectionChange = effect(() => {
+    let selection = this.activeSelection();
+    if (selection) {
+      this.formatSelectionLabel(selection);
```

Comment:
Instead of effect, `computed` signal looks better variant to form `mainPickerLabel`

```
protected mainPickerLabel = computed(() => {
   const selection = this.activeSelection();
   if (!selection) {
      return;
    }
    if (selection.type === 'FULL' || selection.type === 'ABSOLUTE') {
      const range = selection.absoluteSelection;
      if (range) {
        return TimeSeriesUtils.formatRange(range);
      } else {
        return 'Full range';
      }
    } else {
      return selection.relativeSelection!.label!;
    }
})
```